packages/components/src/PieChart/PieChartButton.component.js

```diff
@@ -0,0 +1,156 @@
+ * @param {string} overlayId the id to be set for the overlay
+ * @param {callback} overlayRef the callback to be set for the overlay to bind ref
+ */
+export function decorateWithOverlay(
+	btn,
+	overlayPlacement,
+	overlayComponent,
+	overlayId,
+	overlayRef,
+) {
+	if (!overlayComponent) {
+		return btn;
+	}
+	return (
+		<span>
+			<OverlayTrigger
+				trigger="click"
+				rootClose
+				placement={overlayPlacement}
+				overlay={<Popover id={overlayId}>{overlayComponent}</Popover>}
+				ref={overlayRef}
+			>
+				{btn}
+			</OverlayTrigger>
+		</span>
+	);
+}
+
+/**
+ * This function wrap the button with a TooltipTrigger
+ * @param {Element} btn the button element ( may be wrapped by overlay trigger )
+ * @param {boolean} tooltip tell if the tooltip has to be showed
+ * @param {string} label the label to show on the tooltip
+ * @param {string} tooltipPlacement the tooltip placement
+ */
+export function decorateWithTooltip(btn, tooltip, label, tooltipPlacement) {
+	if (!tooltip || !label) {
+		return btn;
+	}
+	return (
+		<TooltipTrigger label={label} tooltipPlacement={tooltipPlacement}>
+			{btn}
+		</TooltipTrigger>
+	);
+}
+
+/**
+ * This function wrap the event when we don't have an overlay component
+ * @param {function} mouseEvent the event to wrap ( mouseClick )
+ * @param {Element} overlayComponent tell if there is an overlay component
+ * @param {string} label the label of the component
+ * @param {object} rest the rest of the props
+ * @param {object} model the model of the component
+ */
+export function wrapMouseEvent(mouseEvent, overlayComponent, label, rest, model) {
+	if (overlayComponent || !mouseEvent) {
+		return null;
+	}
+	return event =>
+		mouseEvent(event, {
+			action: { label, ...rest },
+			model,
+		});
+}
+
+export default function PieChartButtonComponent({
+	available,
+	buttonRef,
+	className,
+	display,
+	hideLabel,
+	label,
+	labelIndex,
+	loading,
+	minimumPercentage,
+	model,
+	onClick,
+	onMouseDown,
+	overlayComponent,
+	overlayId,
+	overlayPlacement,
+	overlayRef,
+	size,
+	tooltip,
+	tooltipPlacement,
+	...rest
+}) {
+	if (!available) {
+		return null;
+	}
+	const rClick = wrapMouseEvent(onClick, overlayComponent, label, rest, model);
+	const rMouseDown = wrapMouseEvent(onMouseDown, overlayComponent, label, rest, model);
+	let btn = (
+		<Button
+			className={classnames(theme['tc-pie-chart-button'], 'tc-pie-chart-button', className)}
+			onMouseDown={rMouseDown}
+			onClick={rClick}
+			ref={buttonRef}
+			bsStyle="link"
+			role="button"
+			{...rest}
+		>
+			<PieChart
+				display={display}
+				hideLabel={hideLabel}
+				labelIndex={labelIndex}
+				loading={loading}
+				minimumPercentage={minimumPercentage}
+				model={model}
+				size={size}
+			/>
+		</Button>
+	);
+	btn = decorateWithOverlay(btn, overlayPlacement, overlayComponent, overlayId, overlayRef);
+	btn = decorateWithTooltip(btn, tooltip, label, tooltipPlacement);
+	return btn;
```

Comment:
Should return directly.
`return decorateWithTooltip(btn, tooltip, label, tooltipPlacement);`